### PR TITLE
Fix update/add function in Controller & refactor to save function in …

### PR DESCRIPTION
**Pull request by JonathanYesh279**
I fixed the save function: 
first i excluded the id from the boardToSave and then save the board without the id because mongo does not allow it.
i had typo error with the property backgroundImage in the addBoard function inside the controller and changed it .

in the update function it takes all the updates from the req.body and then update the exiting boardwith the neccessary fields to updates from the put request...

### CODE REVIEW

File: api/board/board.controller.js [hunk 1]
```diff
@@ -66,15 +71,13 @@ export async function addBoard(req, res) {
 export async function updateBoard(req, res) {
   try {
     const { boardId } = req.params
-    const boardToInsert = req.body
+    const updates = req.body
 
-    if (!boardId) {
-      return res.status(400).send('Bad request')
-    }
+    const existingBoard = req.board
```
Comment (tomercarolla) [inline api/board/board.controller.js:76]: req.board? I dont think is right.. we send the board updates in the body, but I dont think we send the board in the request.

Comment (tomercarolla) [reply api/board/board.controller.js:76]: you test in the postman?

File: services/db.service.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
-import { MongoClient } from 'mongodb'
+import { MongoClient, ObjectId } from 'mongodb'
```
Comment (tomercarolla) [inline services/db.service.js:1]: you dont use here ObjectIf remove the import

File: api/board/board.controller.js (hunk 1 above)
Comment (JonathanYesh279) [reply api/board/board.controller.js:76]: > req.board? I dont think is right.. we send the board updates in the body, but I dont think we send the board in the request.


We get it from the middleware premission that we have before it gets to the update route.. then we can grab it from there and use it

File: services/db.service.js (hunk 2 above)
Comment (JonathanYesh279) [reply services/db.service.js:1]: Yes, we can remove it

File: api/board/board.controller.js (hunk 1 above)
Comment (JonathanYesh279) [reply api/board/board.controller.js:76]: > you test in the postman?

It suppose to work. It worked with postman

**Review by tomercarolla — APPROVED** [review]

**Pull request merged**